examples/virtual_methods/base_button/imp.rs

```diff
@@ -0,0 +1,103 @@
+use super::BaseButtonExt;
+use gtk::{
+    gio,
+    glib::{self, Error},
+    prelude::*,
+    subclass::prelude::*,
+};
+use std::{future::Future, pin::Pin};
+
+pub type BaseButtonInstance = super::BaseButton;
+pub type PinnedFuture = Pin<Box<dyn Future<Output = Result<(), Error>> + 'static>>;
+
+/// GObject glue code for our BaseButtonClass which holds the function pointers to our virtual functions.
+#[repr(C)]
+pub struct BaseButtonClass {
+    pub parent_class: gtk::ffi::GtkButtonClass,
+    pub sync_method: Option<unsafe fn(&BaseButtonInstance, extra_text: Option<String>)>,
+    pub async_method: Option<unsafe fn(&BaseButtonInstance) -> PinnedFuture>,
+}
+
+unsafe impl ClassStruct for BaseButtonClass {
+    type Type = BaseButton;
+}
+
+impl std::ops::Deref for BaseButtonClass {
+    type Target = glib::Class<glib::Object>;
+
+    fn deref(&self) -> &Self::Target {
+        unsafe { &*(self as *const _ as *const Self::Target) }
+    }
+}
+
+impl std::ops::DerefMut for BaseButtonClass {
+    fn deref_mut(&mut self) -> &mut glib::Class<glib::Object> {
+        unsafe { &mut *(self as *mut _ as *mut glib::Class<glib::Object>) }
```

Comment:
```suggestion
        unsafe { &mut *(self as *mut _ as *mut Self::Target) }
```

Maybe

Reply:
Seems like we don't even need these anymore.